optimade/validator/utils.py

```diff
@@ -214,29 +223,40 @@ def get(self, request: str):
 
         status_code = None
         retries = 0
+        errors = []
         # probably a smarter way to do this with requests, but their documentation 404's...
         while retries < self.max_retries:
             retries += 1
             try:
-                self.response = requests.get(self.last_request, headers=self.headers)
-            except requests.exceptions.ConnectionError as exc:
-                sys.exit(
-                    f"{exc.__class__.__name__}: No response from server at {self.last_request}, please check the URL."
+                self.response = requests.get(
+                    self.last_request,
+                    headers=self.headers,
+                    timeout=(self.timeout, DEFAULT_READ_TIMEOUT),
                 )
+
+                status_code = self.response.status_code
+                if status_code != 429:
+                    return self.response
+
+            # If the connection times out, retry but cache the error
+            except requests.exceptions.ConnectionError as exc:
+                errors.append(str(exc))
+
             except requests.exceptions.MissingSchema:
                 sys.exit(
                     f"Unable to make request on {self.last_request}, did you mean http://{self.last_request}?"
                 )
-            status_code = self.response.status_code
-            if status_code != 429:
-                break
 
+            # If the connection failed, or returned a 429, then wait 1 second before retrying
             time.sleep(1)
+            continue
```

Comment:
Just a minor point, but why is there a continue statement here? 
Are you not already at the end of the while loop?

Reply:
Thanks, this was an unintentional addition